src/arrays.jl

```diff
@@ -23,36 +23,37 @@ function deepdiff(X::Vector, Y::Vector)
     # substrings.
 
     lengths = zeros(Int, length(X)+1, length(Y)+1)
-    backtracks = fill(:nothing, axes(lengths))
+    backtracks = fill(:nothing, length(X), length(Y))
 
     for (j, v2) in enumerate(Y)
         for (i, v1) in enumerate(X)
             if isequal(v1, v2)
                 lengths[i+1, j+1] = lengths[i, j] + 1
-                backtracks[i+1, j+1] = :XY
+                backtracks[i, j] = :XY
             else
-                (lengths[i+1, j+1], backtracks[i+1, j+1]) = _argmax(lengths[i+1, j], lengths[i, j+1])
+                (lengths[i+1, j+1], backtracks[i, j]) = _argmax(lengths[i+1, j], lengths[i, j+1])
             end
         end
     end
 
     removed = Int[]
     added = Int[]
-    backtrack(lengths, removed, added, X, Y, length(X), length(Y))
+
+    backtrack(lengths, backtracks, removed, added, X, Y, length(X), length(Y))
```

Comment:
I feel like this could be cleaned up if we store most of this information in a new struct